### PR TITLE
Add Gateway Rejection Reason to Transaction

### PR DESCRIPTION
**Pull request by ccahoon**
Add Gateway Rejection Reason to Transaction.

### CODE REVIEW

**Comment by ccahoon** [comment]
I'm moving these to gateway_rejection_reason.go. What do you think about the prefix?

File: gateway_rejection_reason.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+const (
+	GatewayRejectionReasonApplicationIncomplete GatewayRejectionReason = "application_incomplete"
+	GatewayRejectionReasonAVS                                          = "avs"
+	GatewayRejectionReasonAVSAndCVV                                    = "avs_and_cvv"
+	GatewayRejectionReasonCVV                                          = "cvv"
+	GatewayRejectionReasonDuplicate                                    = "duplicate"
+	GatewayRejectionReasonFraud                                        = "fraud"
+	GatewayRejectionReasonThreeDSecure                                 = "three_d_secure"
+	GatewayRejectionReasonUnrecognized                                 = "unrecognized"
```
Comment (leighmcculloch) [inline gateway_rejection_reason.go]: The only identifier that is typed in this list is the first one, the others are untyped consts. e.g. https://play.golang.org/p/sK63YlkNXk

File: transaction.go [hunk 2]
```diff
@@ -69,13 +70,10 @@ type TransactionRequest struct {
 // TODO: not all transaction fields are implemented yet, here are the missing fields (add on demand)
 //
 // <transaction>
-//   <custom-fields>
-//   </custom-fields>
```
Comment (leighmcculloch) [inline transaction.go]: Cheers, thanks!

File: transaction_integration_test.go [hunk 3]
```diff
@@ -241,6 +241,10 @@ func TestTransactionCreateWhenGatewayRejectedFraud(t *testing.T) {
 		t.Fatalf("Got status %q, want %q", txn.Status, "gateway_rejected")
 	}
 
+	if txn.GatewayRejectionReason != GatewayRejectionReasonFraud {
+		t.Fatalf("Got gateway rejection reason '%s', wanted 'fraud'", txn.GatewayRejectionReason)
```
Comment (leighmcculloch) [inline transaction_integration_test.go]: If you replace `'%s'` with `%q` you'll get a quoted string out the other side, which is consistent with the other expectations in this test file. e.g. https://play.golang.org/p/bCG-ipotJv

File: gateway_rejection_reason.go (hunk 1 above)
Comment (ccahoon) [inline gateway_rejection_reason.go]: Nice catch! 

FWIW, I looked around to see what the current state of things was to mimic it-- do you think it's worth a potentially breaking change to go around and update those for consistency? main.go, for example, has the bug you pointed out here (only one of the constants is typed). subscription.go has the same convention format as this, but has a string type. That kind of thing. disbursement.go does not have the same naming convention as this and is untyped.

Comment (leighmcculloch) [inline gateway_rejection_reason.go]: I've been meaning to fix up those other enums. I think we should fix them all and make them all typed. It's a breaking change but rather minor break, and until we reach v1 we've been allowing breaking changes.

**Comment by ccahoon** [comment]
#143

**Pull request merged**